lib/ruby_lsp/executor.rb

```diff
@@ -212,15 +212,25 @@ def show_syntax_tree(uri, range)
       { ast: Requests::ShowSyntaxTree.new(@store.get(uri), range).run }
     end
 
-    sig { params(uri: URI::Generic, position: Document::PositionShape).returns(T.nilable(Interface::Location)) }
+    sig do
+      params(
+        uri: URI::Generic,
+        position: Document::PositionShape,
+      ).returns(T.nilable(T.any(T::Array[Interface::Location], Interface::Location)))
```

Comment:
Wouldn't it be easier to always return a nilable array even if it contains only one location?

Reply:
The spec allows you to return either an array of locations or a single one. I don't have a strong opinion, but I liked the implementation better like this.